### PR TITLE
Issue#114_airon [FEATURE REQUEST] CommentsController - GET /comments: Create controller method to get all comments

### PR DESCRIPTION
**Pull request by AironDev**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**ISSUE #114 ?**
[FEATURE REQUEST] "CommentsController - GET /comments: Create controller method to get all comments
I created a single configurable  route to get all comments as specified in the docs

**HOW TO TEST**
- As specified in docs, GET commente/get_comments
- All query parameters specified in docs can be used to configure the endpoint
- isFlagged, refId, origin etc.

**LINK TO GITHUB ISSUE**
[Issue #114 ](https://github.com/microapidev/comment-microapi/issues/114)

**Questions:**

**SLACK DETAILS**
- username: @airon
- email: airondev@gmail.com
- full-name: Aaron, Aniebiet B.

### CODE REVIEW

**Comment by AironDev** [comment]
Proposed changes effected -

**Comment by AironDev** [comment]
image

**Comment by AironDev** [comment]
image

refId query parameter specified

File: controller/commentsController.js [hunk 1]
```diff
@@ -167,6 +169,69 @@ exports.flagComment = async (req, res, next) => {
...
+      status: "error",
+      message: `Something went wrong`,
+      data: err,
+    });
+  }
+};
+// issue#114_airon ends
+
+// Nothing is happening here for now - just to avoid lint errors
+exports.unusedMethod = async () => {
```
Comment (dave-ok) [inline controller/commentsController.js:228]: Please remove this function before pushing your changes, since it is of no use

**Comment by AironDev** [comment]
done @dave-ok

File: controller/commentsController.js [hunk 2]
```diff
@@ -167,6 +169,69 @@ exports.flagComment = async (req, res, next) => {
...
+            },
+            _id: comments[i]._id,
+            refId: comments[i].refId,
+            applicationId: comments[i].applicationId,
+            ownerId: comments[i].ownerId,
+            content: comments[i].content,
+            origin: comments[i].origin,
+            createdAt: comments[i].createdAt,
+            updatedAt: comments[i].updatedAt,
+            __v: comments[i].__v,
```
Comment (simply-alliv) [inline controller/commentsController.js:206]: Please ensure that this is what you return. If a field is not in the swagger docs do not add it until its there. The current version of swagger docs is unavailable on the live server but if you pull the latest updates to your repo, you will find that the comment response is expected to look like this. We will add other fields and properties later. We just want to have a functional product right now.

```
 {
  "message": "string",
  "response": "string",
  "data": [
    {
      "commentId": "string",
      "refId": "string",
      "applicationId": "string",
      "ownerId": "string",
      "content": "string",
      "origin": "string",
      "numOfVotes": 0,
      "numOfUpVotes": 0,
      "numOfDownVotes": 0,
      "numOfFlags": 0,
      "numOfReplies": 0
    }
  ]
}
```

File: controller/commentsController.js [hunk 3]
```diff
@@ -167,6 +169,69 @@ exports.flagComment = async (req, res, next) => {
...
+          });
+        }
+        res.status(200).json({
+          status: "success",
+          message: `Comments Retrieved Successfully for ${applicationId}/${refId}`,
+          query: query,
+          data: data,
+        });
```
Comment (simply-alliv) [inline controller/commentsController.js:214]: Please use the responseHandler from the utils folder. See other controller implementations for examples.

File: controller/commentsController.js [hunk 4]
```diff
@@ -167,6 +169,69 @@ exports.flagComment = async (req, res, next) => {
...
+      })
+      .catch(next);
+  } catch (err) {
+    res.status(401).json({
+      status: "error",
+      message: `Something went wrong`,
+      data: err,
+    });
```
Comment (simply-alliv) [inline controller/commentsController.js:222]: Please use the CustomError from the utils folder. See other controller implementations for examples.

File: controller/commentsController.js (hunk 3 above)
Comment (AironDev) [reply controller/commentsController.js]: done

**Pull request merged**